pkg/proxyagent/agent/agent.go

```diff
@@ -194,6 +199,67 @@ func GetClusterProxyValueFunc(
 		if err != nil {
 			return nil, err
 		}
+
+		// Get agentIndentifiers and servicesToExpose.
+		// agetnIdentifiers is used in `--agent-identifiers` flag in addon-agent-deployment.yaml.
+		// servicesToExpose defines the services we want to expose to the hub.
+
+		// List all available managedClusterSets
+		managedClusterSetList := &clusterv1beta1.ManagedClusterSetList{}
+		err = runtimeClient.List(context.TODO(), managedClusterSetList)
+		if err != nil {
+			return nil, err
+		}
+
+		// deleted managedClusterSets are not included in the list
+		managedClusterSets := filterManagedClusterSetByDeletionTimestamp(managedClusterSetList.Items)
+
+		// only cluseterSet cover current cluster include in the list.
+		managedClusterSets, err = filterManagedClusterSetForCurrentCluster(managedClusterSets, cluster.Labels)
+		if err != nil {
+			return nil, err
+		}
+
+		managedClusterSetMap := map[string]clusterv1beta1.ManagedClusterSet{}
+		for _, managedClusterSet := range managedClusterSets {
```

Comment:
could we use one loop to handle this?

```
for _,  managedClusterSets := range managedClusterSetList.Items {
		if !managedClusterSet.DeletionTimestamp.IsZero() {
			continue
		}
                selector, err := clusterv1beta1.BuildClusterSelector(&managedClusterSet)
		if err != nil {
			return nil, err
		}
		if !selector.Matches(labels.Set(clusterlabels)) {
			continue
		}

               managedClusterSetMap[managedClusterSet.Name] = managedClusterSet
               
}
```

Reply:
Yes, we can. but here is a trade-off for unit-test sake.

Reply:
you can put this loop in one function, I think it is still easy to test with unit

Reply:
Done.